### PR TITLE
make MaxTxGasWanted configurable

### PR DESCRIPTION
**Pull request by yihuang**
## Description

Closes: #1003

Solution:
- add config item: evm.max-tx-gas-wanted

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/ante/eth.go [hunk 1]
```diff
@@ -188,8 +191,8 @@ func (egcd EthGasConsumeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 
 		if ctx.IsCheckTx() {
 			// We can't trust the tx gas limit, because we'll refund the unused gas.
-			if txData.GetGas() > MaxTxGasWanted {
-				gasWanted += MaxTxGasWanted
+			if txData.GetGas() > egcd.maxGasWanted {
+				gasWanted += egcd.maxGasWanted
```
Comment (fedekunze) [inline app/ante/eth.go:195]: can this cause non-determinism if nodes use different values?

Comment (yihuang) [reply app/ante/eth.go:195]: this is only for check tx, so I think it's ok.

**Review by fedekunze — APPROVED** [review]

**Comment by zmanian** [comment]
This doesn't really solve the issue.

On evmos right now, 80% of tx volume is logic of " test if arb is profitable, if profitable arb otherwise revert". This results in a 7x ratio between gas wanted and gas used which then results in full blocks.

This is what we originally discussed for this scenario.

https://github.com/cosmos/cosmos-sdk/issues/2150

Gas should be charged at max(gas_used, gas_wanted/2)

You may want to just do this in the ethermint ante handler.

**Comment by yihuang** [comment]
> Gas should be charged at max(gas_used, gas_wanted/2)

Do you mean only refund `gas_wanted - max(gas_used, gas_wanted/2)` at the end of tx processing?

**Comment by yihuang** [comment]
> This doesn't really solve the issue.
> 
> On evmos right now, 80% of tx volume is logic of " test if arb is profitable, if profitable arb otherwise revert". This results in a 7x ratio between gas wanted and gas used which then results in full blocks.
> 
> This is what we originally discussed for this scenario.
> 
> [cosmos/cosmos-sdk#2150](https://github.com/cosmos/cosmos-sdk/issues/2150)
> 
> Gas should be charged at max(gas_used, gas_wanted/2)
> 
> You may want to just do this in the ethermint ante handler.

https://github.com/tharsis/ethermint/issues/1085

I opened an issue here following your suggestion, summarize the consequences of this as I understand.

**Pull request merged**